frontend/__tests__/src/components/metrics/ConfigStep/SourceControl.test.tsx

```diff
@@ -0,0 +1,123 @@
+  MOCK_SOURCE_CONTROL_URL,
+  RESET,
+  SOURCE_CONTROL_FIELDS,
+  SOURCE_CONTROL_TYPES,
+  VERIFIED,
+  VERIFY,
+} from '../../../fixtures'
+import { setupServer } from 'msw/node'
+import { rest } from 'msw'
+
+export const fillSourceControlFieldsInformation = () => {
+  const mockInfo = 'mockToken'
+  const tokenInput = screen.getByTestId('sourceControlTextField').querySelector('input') as HTMLInputElement
+  fireEvent.change(tokenInput, { target: { value: mockInfo } })
+  expect(tokenInput.value).toEqual(mockInfo)
+}
+
+let store = setupStore()
+const setup = () => {
+  store = setupStore()
+  return render(
+    <Provider store={store}>
+      <SourceControl />
+    </Provider>
+  )
+}
+
+const server = setupServer(
+  rest.get(MOCK_SOURCE_CONTROL_URL, (req, res, ctx) => {
+    return res(ctx.status(200))
+  })
+)
+
+describe('SourceControl', () => {
+  beforeAll(() => server.listen())
+  afterAll(() => server.close())
+  it('should show sourceControl title and fields when render sourceControl component', () => {
+    const { getByRole, getByLabelText } = setup()
+
+    expect(getByRole('heading', { name: CONFIG_TITLE.SOURCE_CONTROL })).toBeInTheDocument()
+    SOURCE_CONTROL_FIELDS.map((field) => {
+      expect(getByLabelText(`${field} *`)).toBeInTheDocument()
+    })
+  })
+
+  it('should show default value gitHub when init sourceControl component', () => {
+    const { getByText } = setup()
+    const sourceControlType = getByText(SOURCE_CONTROL_TYPES.GIT_HUB)
+
+    expect(sourceControlType).toBeInTheDocument()
+  })
+
+  it('should clear all fields information when click reset button', async () => {
+    const { getByRole, getByText, queryByRole } = setup()
+    const tokenInput = screen.getByTestId('sourceControlTextField').querySelector('input') as HTMLInputElement
+    fillSourceControlFieldsInformation()
+
+    fireEvent.click(getByText(VERIFY))
+
+    await waitFor(() => {
+      expect(getByRole('button', { name: RESET })).toBeInTheDocument()
+      fireEvent.click(getByRole('button', { name: RESET }))
+    })
+
+    expect(tokenInput.value).toEqual('')
+    expect(getByText(SOURCE_CONTROL_TYPES.GIT_HUB)).toBeInTheDocument()
+    expect(queryByRole('button', { name: RESET })).not.toBeTruthy()
+    expect(queryByRole('button', { name: VERIFY })).toBeDisabled()
+  })
+
+  it('should enable verify button when all fields checked correctly given disable verify button', () => {
+    const { getByRole } = setup()
+    const verifyButton = getByRole('button', { name: VERIFY })
+
+    expect(verifyButton).toBeDisabled()
+
+    fillSourceControlFieldsInformation()
+    expect(verifyButton).toBeEnabled()
+  })
+
+  it('should show reset button  and verified button when verify successfully', async () => {
+    const { getByText } = setup()
+    fillSourceControlFieldsInformation()
+    fireEvent.click(getByText(VERIFY))
+    await waitFor(() => {
+      expect(getByText(RESET)).toBeVisible()
+    })
+    await waitFor(() => {
```

Comment:
same issue   
toBeVisible or toBeTruthy

Reply:
maybe change it to `toBeInTheDocument`, it is better to make sense

Reply:
use uniform matcheres